src/vcf/record.cpp

```diff
@@ -472,6 +505,19 @@ namespace ebi
         }
     }
 
+    void Record::check_no_duplicates(std::vector<std::string> const & values) const
+    {
+        if (values.size() > 1) {
+            std::map<std::string, int> counter;
+            for (auto & value : values) {
+                counter[value]++;
+                if (counter[value] >= 2) {
+                    throw std::invalid_argument(" must not have duplicate ");
```

Comment:
This message is unclear just by itself. I think it would be best to throw the exception without any message, then build the whole message in the methods outside. Applies to ID, FILTER and FORMAT that I can see.